Feat: Обновил перегруженые методы process. Теперь нет формирований "л…

TinalexD opened the pull request:
…ишних" пустых сообщений. Добавил Исключения в проверку валидности агрументов

TinalexD committed:
Feat: Обновил перегруженые методы process. Теперь нет формирований "лишних" пустых сообщений. Добавил Исключения в проверку валидности агрументов

TinalexD committed:
Feat: Добавил проверку исключений, когда последний элемент пустой. Убрал теперь ненужные IF'ы. Поменял ValidatedService на void, т.к. нам по сути больше не нужен ответ true/false

SvetlanaUtina commented on src/main/java/com/tcs/edu/decorator/OrderedDistinctedMessageService.java:
```diff
@@ -44,13 +45,16 @@ public void process(Message message, Message... massages) {
         for (Message currentMessage : listOfMessages) {
-            if (super.isArgsValid(currentMessage)) {
+            try {
+                super.isArgsValid(currentMessage);
```
я думаю достаточно только эту строку завернуть в блок try catch

SvetlanaUtina replied:
я думаю достаточно только эту строку завернуть в блок try catch

SvetlanaUtina commented on src/main/java/com/tcs/edu/decorator/OrderedDistinctedMessageService.java:
```diff
@@ -61,28 +65,34 @@ public void process(Message message, Message... massages) {
-                if (!listOfDublicates.contains(currentMessage.getMessage())) {
-                    listOfDublicates.add(currentMessage.getMessage());
+            try {
+                super.isArgsValid(message);
```
думаю это лишнее, ибо этот метод убирает дубликаты сообщений и тут не должно быть проверки валидности.

SvetlanaUtina replied:
тогда и проверка сохраняется

SvetlanaUtina commented on src/main/java/com/tcs/edu/decorator/OrderedDistinctedMessageService.java:
```diff
@@ -61,28 +65,34 @@ public void process(Message message, Message... massages) {
-                if (!listOfDublicates.contains(currentMessage.getMessage())) {
-                    listOfDublicates.add(currentMessage.getMessage());
+            try {
+                super.isArgsValid(message);
```
думаю это лишнее, ибо этот метод убирает дубликаты сообщений и тут не должно быть проверки валидности.

SvetlanaUtina commented on src/main/java/com/tcs/edu/decorator/OrderedDistinctedMessageService.java:
```diff
@@ -61,28 +65,34 @@ public void process(Message message, Message... massages) {
+                int filterCount = 0;
+                for (int i = 0; i < sortedMessages.length - 1; i++) {
+                    super.isArgsValid(sortedMessages[i]);
+                    if (!listOfDublicates.contains(sortedMessages[i].getMessage())) {
```
кстати я вот не обратила внимание на это сравнение. скорее всего надо сравнивать может объекты Message, а не только текстовку там же еще северити.

SvetlanaUtina replied:
если так делать, то необходимо переписать еще и сортировку, а в тексте задания не сказано что мы берем в условие еще и Severity, так что сравнение лучше по текст из объекта Message

SvetlanaUtina commented on src/main/java/com/tcs/edu/decorator/OrderedDistinctedMessageService.java:
```diff
@@ -61,28 +65,34 @@ public void process(Message message, Message... massages) {
+                    }
+                }
+
+                Message[] filteredMessages = new Message[listOfMassages.size()];
```
можно listOfMassages.stream().toArray(Message[]::new)

TinalexD committed:
Feat: Заменил рудиментный сбор массива с проходом по списку на преобразование списка в массив

TinalexD committed:
Feat: Замени использование интерфейса на MessageRepository с методом create.
Реализовал in-memory хранение сообщений
Обогатил объект Message переменными id и decoratedMassage. Согласно сообщению из слака можно реализовать вариант, когда будет сохраняться декорированное сообщение

TinalexD committed:
Feat: Добавил операцию Read по первичному ключу

TinalexD committed:
Feat: Добавил операцию Read по всем сообщениям

TinalexD committed:
Feat: Добавил операцию Read по сообщениям с определенным уровнем Severity

TinalexD committed:
Feat: Добавил ID Message в методы equals и hashCode

TinalexD committed:
Feat: Добавил ID Message в метод toString

TinalexD committed:
Feat: Удалил лишний интерфейс, добавил messageRepository в конструктор.

TinalexD committed:
Merge pull request #11 from TinalexD/dev2

Dev2
